### PR TITLE
Feat/normal map compression

### PR DESCRIPTION
**Pull request by Atlante45**
Adds support for compressed normal maps.

 ## Test plan:
- Go to `Developer > Graphics...`
- Turn on "Compress Normal Textures"
- Shutdown Interface
- Clean cache
- Verify normal maps are correct. (Wooden floor, coffee cups in Welcome, Rock floor in playa, sand in localhost/garden)
- Standard graphics smoke test
(Ping me on Slack so I can guide you through doing that if needed)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c24e697088db6189501b89feea4c25913075dc85 available at: https://gist.github.com/a456540cdd0dc1bfb024b0d540936477<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-c24e697.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-c24e697.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e6d66ff7908f4a81afd1080c09b116d83b6c6723 available at: https://gist.github.com/a456540cdd0dc1bfb024b0d540936477<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-e6d66ff.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-e6d66ff.dmg

**Comment by Atlante45** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e6d66ff7908f4a81afd1080c09b116d83b6c6723 available at: https://gist.github.com/a456540cdd0dc1bfb024b0d540936477<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-e6d66ff.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-e6d66ff.dmg

File: libraries/render-utils/src/MaterialTextures.slh [hunk 1]
```diff
@@ -64,7 +64,9 @@ float fetchRoughnessMap(vec2 uv) {
 uniform sampler2D normalMap;
 vec3 fetchNormalMap(vec2 uv) {
     // unpack normal, swizzle to get into hifi tangent space with Y axis pointing out
-    return normalize(texture(normalMap, uv).rbg -vec3(0.5, 0.5, 0.5));
+    vec2 t = 2.0 * (texture(normalMap, uv).rg - vec2(0.5, 0.5));
+    vec2 t2 = t*t;
+    return normalize(vec3(t.x, sqrt(1 - t2.x - t2.y), t.y));
```
Comment (samcake) [inline libraries/render-utils/src/MaterialTextures.slh]: I think the normalize is not needed there anymore

**Comment by samcake** [comment]
@Atlante45 You also need to change the code path when generating the normal map from the bump map,

File: libraries/render-utils/src/MaterialTextures.slh (hunk 1 above)
Comment (samcake) [inline libraries/render-utils/src/MaterialTextures.slh]: To make sure you could do it on only half of the screen (using gl_FragCoord.x > 500 for example) and look with the debug view and make sure there is no difference

**Comment by Atlante45** [comment]
@samcake It's the same code path. I unified all the similar textures (Color, Grayscale, Normal Map) when I worked on the image library.

File: libraries/render-utils/src/MaterialTextures.slh (hunk 1 above)
Comment (Atlante45) [inline libraries/render-utils/src/MaterialTextures.slh]: Here is what I used:
```glsl
    if (gl_FragCoord.y > 400) {
        if (gl_FragCoord.x > 500 && gl_FragCoord.x < 900) {
            return vec3(1.0, 0.0, 0.0);
        } else {
            return vec3(t.x, sqrt(1 - t2.x - t2.y), t.y);
        }
    } else {
        return normalize(vec3(t.x, sqrt(1 - t2.x - t2.y), t.y));    
    }
```
(Square in the center is to identify the break line)

**Comment by hifi-gustavo** [comment]
Build console output for commit: f728ae795595550182cc45554f85f020a2a5caa6 available at: https://gist.github.com/a456540cdd0dc1bfb024b0d540936477<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-f728ae7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-f728ae7.dmg

**Comment by Atlante45** [comment]
@samcake I removed "Dev Approved" since I made some changes.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ecda3132233f36dd53046ab9ff92f730fd0adc0e available at: https://gist.github.com/a456540cdd0dc1bfb024b0d540936477<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-ecda313.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-ecda313.dmg

**Comment by claudioshg** [comment]
testing

**Comment by claudioshg** [comment]
### Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested with Oculus Rift

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10366-ecda313.exe

PASS:
- All map textures are visually correct. The wooden floor, coffee cups in Welcome, Rock floor in playa, sand in localhost/garden.

**Comment by cozza13** [comment]
@samcake dev approves this logged in as @cozza13

**Pull request merged**